lib/tools/adb-commands.js

```diff
@@ -1002,7 +1002,7 @@ methods.killProcessesByName = async function (name) {
  */
 methods.killProcessByPID = async function (pid) {
   log.debug(`Attempting to kill process ${pid}`);
-  return await this.shell(['kill', pid]);
+  return await this.shell(['kill', '-9', pid]);
```

Comment:
it might be useful to try "normal" kill first. Then make sure the process is killed by invoking kill with -0 signal and only after that send SIGKILL if the process still exists after the timeout. _waitForCondition_ helper method from asyncbox module could be also handy.

Reply:
yeah, that will be better to try KILL PID first, and check whether uiautomator process is killed or not. 
maybe you can change the code and pull a new request? Because I am not familiar to Nodejs

Reply:
done in https://github.com/appium/appium-adb/pull/262

Reply:
Much thanks!